cmds/core/uptime/uptime.go

```
@@ -41,23 +47,33 @@ func uptime(contents string) (*time.Time, error) {
 	return &uptime, nil
 }
 
-func main() {
-	procUptimeOutput, err := os.ReadFile("/proc/uptime")
+func run(stdout io.Writer, uptimePath, loadavgPath string) error {
+	procUptimeOutput, err := os.ReadFile(uptimePath)
 	if err != nil {
-		log.Fatalf("error reading /proc/uptime: %v \n", err)
+		return fmt.Errorf("error reading /proc/uptime: %w", err)
 	}
+
 	uptimeTime, err := uptime(string(procUptimeOutput))
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
-	procLoadAvgOutput, err := os.ReadFile("/proc/loadavg")
+
+	procLoadAvgOutput, err := os.ReadFile(loadavgPath)
 	if err != nil {
-		log.Fatalf("error reading /proc/loadavg: %v \n", err)
+		return fmt.Errorf("error reading /proc/loadavg: %w", err)
```

same here, I bet you can return the err all by itself